File: BE-Task_1-Php/internal/repository/memory/purchase_repository.php
<?php

declare(strict_types=1);

namespace App\Internal\Repository\Memory;

use App\Internal\Domain\Entities\Purchase;
use App\Internal\Repository\Interfaces\PurchaseRepositoryInterface;

final readonly class PurchaseRepository implements PurchaseRepositoryInterface
{
    public function __construct(private Store $store)
    {
    }

    public function findById(string $purchaseId): ?Purchase
    {
        return $this->store->purchases[$purchaseId] ?? null;
    }

    public function listAll(): array
    {
        return array_values($this->store->purchases);
    }
}

File: BE-Task_1-Ruby/internal/repository/memory/purchase_repository.rb
module RefundBackend
  module Repository
    module Memory
      class PurchaseRepository < RefundBackend::Repository::Interfaces::PurchaseRepository
        def initialize(store)
          @store = store
        end

        def find_by_id(purchase_id)
          @store.fetch_purchase(purchase_id)
        end

        def list_all
          @store.list_purchases
        end
      end
    end
  end
end
